lib/pbench/server/api/resources/__init__.py

```diff
@@ -24,7 +24,23 @@
 from pbench.server.utils import UtcTimeHelper
 
 
-class UnauthorizedAccess(Exception):
+class APIAbort(Exception):
+    """
+    Used to report an error and abort if there is a failure in processing of API request.
+    """
+
+    def __init__(self, http_status: int, message: str = None):
+        self.http_status = http_status
+        self.message = message if message else HTTPStatus(http_status).phrase
+
+    def __repr__(self) -> str:
+        return f"API error {self.http_status} : {self.message}"
```

Comment:
I wouldn't; not here, anyway. I'm not really even sure this will ever be used, but I didn't want to completely discard Siddardh's message format.

I suppose if I was going to change anything, perhaps I should have used `str(self)` instead of `{self.message}`, which would work for any subclass. Hmm. ☹️